Client/Frontend/Settings/AppSettingsOptions.swift

```diff
@@ -793,9 +793,13 @@ class SendAnonymousUsageDataSetting: BoolSetting {
             settingDidChange: {
                 AdjustHelper.setEnabled($0)
                 Glean.shared.setUploadEnabled($0)
+                Experiments.setTelemetrySetting($0)
                 Experiments.shared.resetTelemetryIdentifiers()
```

Comment:
nit: I'd almost like to see this `resetTelemetryIdentifiers` use moved inside the `setTelemetrySetting()` function. You would have to check for the `false` condition before calling it, but that cleans up an experiment call from a module in the app that shouldn't care about this little detail.

Reply:
Either that, or `resetTelemetryIdentifiers` should trigger the opt-out...